### PR TITLE
JAPI-150 Created HPCCRecordBuilder test and mock objects for dfscient…

### PR DESCRIPTION
**Pull request by jpmcmu**
… implementors

## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [X] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).

## Checklist:
- [X] I have created a corresponding JIRA ticket for this submission
- [X] My code follows the code style of this project.
  - [X] I have applied the Eclipse code-format template provided.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [X] I have read the HPCC Systems CONTRIBUTORS document (https://github.com/hpcc-systems/HPCC-Platform/wiki/Guide-for-contributors).
- [X] The change has been fully tested:
  - [ ] I have performed unit tests to cover my changes.
  - [ ] I have performed system test and covered possible regressions and side effects.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [X] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.

### CODE REVIEW

**Comment by jpmcmu** [comment]
@rpastrana please review

File: dfsclient/src/test/java/org/hpccsystems/dfs/client/DFSRecordBuilderTest.java [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+    }
+
+    @Test
+    public void hpccRecordBuilderTest() throws Exception
+    {
+        try {
+            HPCCRecordBuilder recordBuilder = new HPCCRecordBuilder(testFieldDefinition);
+            testRecordReader.initialize(recordBuilder);
+
+            while (testRecordReader.hasNext()) {
```
Comment (rpastrana) [inline dfsclient/src/test/java/org/hpccsystems/dfs/client/DFSRecordBuilderTest.java]: are there any other HPCCRecbuilder methods we should test?

File: dfsclient/src/test/java/org/hpccsystems/dfs/client/DFSRecordBuilderTest.java [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+
+public class DFSRecordBuilderTest
+{
+    private FieldDef testFieldDefinition = null;
+    private IRecordReader testRecordReader = null;
+
+    @Before
+    public void setup()
+    {
+        try {
```
Comment (rpastrana) [inline dfsclient/src/test/java/org/hpccsystems/dfs/client/DFSRecordBuilderTest.java]: let's apply the format template to the new code

File: dfsclient/src/test/java/org/hpccsystems/dfs/client/DFSRecordBuilderTest.java (hunk 1 above)
Comment (jpmcmu) [reply dfsclient/src/test/java/org/hpccsystems/dfs/client/DFSRecordBuilderTest.java]: the TestRecordReader will call all of the HpccRecordBuilder functions

**Pull request merged**